docs/esterni/manuale_installazione/manuale_installazione.tex

```diff
@@ -0,0 +1,285 @@
+Una volta eseguito lo script, verrà ritornato e stampato l'id e il prezzo del prodotto messo in vendita.
+\subsubsection{Script python}
+È possibile realizzare un proprio script in python per la vendita di prodotti.\\
+La funzione per mettere in vendita il proprio prodotto è \textbf{sell\_item} di \textit{sell.py}. La funzione richiede in input un solo parametro, ovvero il prezzo in dollari del prodotto che si desidera vendere.\\
+È dunque necessario includere nel proprio script la funzione come segue:
+\begin{verbatim}
+from sell import sell_item
+\end{verbatim}
+\subsection{Test}
+Per eseguire il test sullo script è sufficiente eseguire il seguente comando:
+\begin{lstlisting}[language=bash]
+$ pytest
+\end{lstlisting}
+\section{Contract}
+\subsection{Requisiti}
+\subsubsection{Truffle, plugin e librerie}
+Per compilare lo smart contract e deployarlo è necessario installare Truffle e alcuni plugin e librerie all'interno della cartella "Contract". Eseguire i seguenti comandi per una corretta installazione:
+\begin{lstlisting}[language=bash]
+  $ npm install
+  $ npm install -g truffle
+  $ npm install @truffle/hdwallet-provider
+\end{lstlisting}
+per le librerie eseguire:
+\begin{lstlisting}[language=bash]
+  $ npm install @openzeppelin/contracts
+  $ npm install @chainlink/contracts
+\end{lstlisting}
+infine per la verifica e test installare:
+\begin{lstlisting}[language=bash]
+  $ npm install truffle-plugin-verify
+  $ npm install solidity-coverage
+\end{lstlisting}
+\subsection{Configurazione}
+È necessaria la creazione di tre file nella cartella "Contract" per il deployment del contratto:
+\begin{itemize}
+  \item \textbf{mnemonic.secret}: deve contenere la mnemonic del proprio wallet;
+  \item \textbf{providerlink.secret}: deve contenere l'url del provider ottenibile come spiegato al punto \hyperref[sec:provider]{2.1.2}, può essere utilizzato anche l'url in http;
+  \item \textbf{apikey.secret}: deve contenere le api key ottenibili come spiegato al punto \hyperref[sec:provider]{2.1.2}.
+\end{itemize}
+\subsection{Compilazione, deployment e verifica}
+\paragraph{Compilazione}\\
+Per la compilazione del contract è sufficiente eseguire il seguente comando:
+\begin{lstlisting}[language=bash]
+  $ truffle compile
+\end{lstlisting}
+\paragraph{Deployment}\\
+Per il deployment del contratto eseguire il seguente comando:
+\begin{lstlisting}[language=bash]
+  $ truffle deploy --netowrk <YOUR_NETWORK>
+\end{lstlisting}
+dove \verb|<YOUR_NETWORK>| rappresenta il network che si desidera utilizzare, in questo caso viene utilizzato \verb|polygon_mumbai|. Il network è definito nel file \textit{truffle-config.js}.\\ \\
+In caso di errori di timeout cambiare l'url nel file \textit{providerlink.secret} da http a wss e/o eseguire il comando aggiungendo:
+\begin{lstlisting}[language=bash]
+  $ truffle deploy --netowrk <YOUR_NETWORK> --reset --compile-none
+\end{lstlisting}
```

Comment:
Andrebbe messo come collegare l'upkeeper di chainlink (trovi in fondo al readme del server mooolto abbozzato)